### DISCUSSION TITLE
Solution filter returning errors which instruct to use desktop msbuild

### DISCUSSION THREAD

**Discussion by cbra2836** [unoplatform/uno#12604] [Q&A]
I am using Microsoft Visual Studio 17.6.2 on a Windows 11 x64 laptop.

I am getting the below error multiple times, when trying to use the solution file for my selected target framework (Windows). Using dotnet msbuild or specifying the option for a framework or target framework (through properties) is not working either.

```
C:\Users\celin\.nuget\packages\msbuild.sdk.extras\3.0.44\Build\Workarounds.targets(27,5): error : If you are building projects that require targets from full MSBuild or MSBuildFrameworkToolsPath, you need to use desktop msbuild ('msbuild.exe') instead of 'dotnet build' or 'dotnet msbuild' [C:\uno\src\AddIns\Uno.UI.MSAL\Uno.UI 
.MSAL.csproj::TargetFramework=uap10.0.18362]
```

I think I have been able to set up everything correctly except for the loading of some project files, so I suspect these two issues are related. I tried implementing the fix for "Building the reference assemblies for Skia and WebAssembly" in the troubleshooting build issues section of the docs to try solve this problem but it generates the following error:

```
C:\uno\src\T4Generator\T4Generator.csproj : Solution file error MSB5028: Solution filter file at "C:\uno\src\U no.UI-Reference-only.slnf" includes project "T4Generator\T4Generator.csproj" that is not in the solution file at "C:\uno\src\Uno.UI.sln".
```

**Comment by jeromelaban** [comment]
Thanks for the report. On Windows, it's currently not possible to build the solution using `dotnet build`, you'll need to use `msbuild` directly. 

I'm wondering, why do you build using the command line and not visual studio directly?
